lang/cs/Org.Apache.REEF.Examples/DriverRestart/HelloRestartDriver.cs

```diff
@@ -0,0 +1,181 @@
+        {
+        }
+
+        /// <summary>
+        /// Submits the HelloRestartTask to the Evaluator.
+        /// </summary>
+        public void OnNext(IAllocatedEvaluator allocatedEvaluator)
+        {
+            var taskConfiguration = TaskConfiguration.ConfigurationModule
+                .Set(TaskConfiguration.Identifier, "HelloRestartTask")
+                .Set(TaskConfiguration.Task, GenericType<HelloRestartTask>.Class)
+                .Set(TaskConfiguration.OnMessage, GenericType<HelloRestartTask>.Class)
+                .Build();
+            allocatedEvaluator.SubmitTask(taskConfiguration);
+        }
+
+        /// <summary>
+        /// Called to start the driver.
+        /// </summary>
+        public void OnNext(IDriverStarted driverStarted)
+        {
+            _Logger.Log(Level.Info, string.Format("HelloRestartDriver started at {0}", driverStarted.StartTime));
+            for (var i = 0; i < AmountOfTasksToSubmit; i++)
+            {
+                this._evaluatorRequestor.Submit(new EvaluatorRequest(1, 64));
+            }
+        }
+
+        /// <summary>
+        /// Prints a restart message and enters the restart codepath.
+        /// </summary>
+        public void OnNext(IDriverRestarted value)
+        {
+            this._restarted = true;
+            _Logger.Log(Level.Info, "Hello! HelloRestartDriver has restarted!");
+        }
+
+        public void OnNext(IActiveContext value)
+        {
+            _Logger.Log(Level.Info, "Received active context " + value.Id + " from evaluator with ID [" + value.EvaluatorId + "].");
+        }
+
+        public void OnNext(IRunningTask value)
+        {
+            lock (_lockObj)
+            {
+                this._runningTaskCount++;
+                
+                _Logger.Log(Level.Info, "Received running task with ID [" + value.Id + "] " +
+                              " with restart set to " + _restarted + " from evaluator with ID [" + 
+                              value.ActiveContext.EvaluatorId + "]");
+
+                if (_restarted)
+                {
+                    value.Send(Encoding.UTF8.GetBytes("Hello from driver!"));
+                }
+
+                // Kill itself in order for the driver to restart it.
+                if (this._runningTaskCount == AmountOfTasksToSubmit)
+                {
+                    if (_restarted)
+                    {
+                        _Logger.Log(Level.Info, "Retrieved all running tasks from the previous instance.");
+                    }
+                    else
+                    {
+                        Process.GetCurrentProcess().Kill();
+                    }
+                }
+            }
+        }
+
+        public void OnNext(IDriverRestartCompleted value)
+        {
+            _Logger.Log(Level.Info, "Driver restart has completed.");
+        }
+
+        public void OnNext(ICompletedTask value)
+        {
+            IncrementFinishedTask();
```

Comment:
Wouldn't you also need to close the underlying context / evaluator? Otherwise, how do they get shut down?